### DISCUSSION TITLE
在 tsx 中使用 el-icon 时， 进行 onClick 绑定时 ts提示 不存在 onClick 属性

### DISCUSSION THREAD

**Discussion by pengzhanbo** [element-plus/element-plus#8424] [Q&A]
## 在 tsx 中使用 el-icon 时， 进行 onClick 绑定时 ts提示 不存在 onClick 属性

<img width="773" alt="image" src="https://user-images.githubusercontent.com/16745751/175121039-3bcdd9a5-7c1c-4cbf-af81-db71dbc871c4.png">

**Comment by ycx0303** [comment]
+1

**Reply by tolking** [reply]
由于 [inheritattrs](https://cn.vuejs.org/api/options-misc.html#inheritattrs) 的存在，很多原生属性并没有在组件中声明。这些隐形传递的属性会引起 tsx 中的类型错误

可以使用下面这中临时解决方案

```ts
// env.d.ts
declare module '@vue/runtime-core' {
  interface AllowedComponentProps {
    [key: string]: Any
  }
}
```

如果有更好的解决方案，欢迎讨论

**Reply by WangJincheng4869** [reply]
这么配置之后其他函数都报错了，发现一个邪修写法可以过检。
```tsx
<ElForm
  ref={setFormRef}
  model={form.value}
  labelPosition='top'
  {...{
    onSubmit: (e: SubmitEvent) => {
      e.preventDefault();
      alert('submit');
      return;
    },
  }}
>
</ElForm>
```

**Comment by xuxing409** [comment]
Is there a better solution now?
